templates/flavors/README.md

```diff
@@ -0,0 +1,16 @@
+# Flavors
+
+In `clusterctl` the infrastructure provider authors can provide different type of cluster templates, 
+or flavors; use the --flavor flag to specify which flavor to use; e.g
+```shell
+clusterctl config cluster my-cluster --kubernetes-version v1.16.3 \
+    --flavor external-cloud-provider > my-cluster.yaml
+```
+See [`clusterctl` flavors docs](https://cluster-api.sigs.k8s.io/clusterctl/commands/config-cluster.html#flavors).
+
+This directory contains each of the flavors for CAPZ. Each directory besides `base` will be used to
+create a flavor by running `kustomize build` on the directory. The name of the directory will be
+appended to the end of the cluster-template.yaml, e.g cluster-template-{directory-name}.yaml. That
+flavor can be used by specifying `--flavor {directory-name}`.
```

Comment:
@devigned great PR 👍, 
but just a question, I'm a little confused with the difference between base and default, I can see default adds a machine-deployment but when I run `make generate-flavors` I only get `cluster-template.yaml` and `cluster-template-external-cloud-provider.yaml`  which are not corresponding to the directories names, so I was wondering if we can combine base and default or clarify it in the readme a little bit more

Reply:
You are too kind. Thank you.

`base` is ignored currently as a place to hold shared structures. I hope that it goes away in a future PR where we would take the `./templates/base/cluster-template.yaml` and break it up into the individual resources to be used piece by piece to build new flavors. Just thought it would be best to break that change into a couple of PRs.

What do you think?